packages/core/src/validate/run.ts

```diff
@@ -629,6 +661,134 @@ function* getInternalFunctionStorageErrors(
   }
 }
 
+/**
+ * Reports an error if a parent contract has an initializer and any of the following are true:
+ * - 1. Missing initializer: This contract does not appear to have an initializer.
+ * - 2. Missing initializer call: This contract's initializer is missing a call to a parent initializer.
+ * - 3. Duplicate initializer call: This contract has duplicate calls to the same parent initializer function.
+ * - 4. Incorrect initializer order: This contract does not call parent initializers in the correct order.
+ */
+function* getInitializerErrors(
+  contractDef: ContractDefinition,
+  deref: ASTDereferencer,
+  decodeSrc: SrcDecoder,
+): Generator<ValidationErrorInitializer> {
+  if (contractDef.baseContracts.length > 0) {
+    const baseContractDefs = contractDef.baseContracts.map(base =>
+      deref('ContractDefinition', base.baseName.referencedDeclaration),
+    );
+    const baseContractsInitializersMap = new Map(
+      baseContractDefs.map(base => [base.name, getPossibleInitializers(base, true)]),
+    );
+    const baseContractsWithInitializers = baseContractDefs
+      .filter(base => hasInitializers(base.name, baseContractsInitializersMap))
```

Comment:
This should work, and would remove the need for `hasInitializers`
```suggestion
      .filter(base => baseContractsInitializersMap.get(base.name)?.length)
```